src/nvim/eval/funcs.c

```diff
@@ -0,0 +1,106 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check
+// it. PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+#include <stdio.h>
+#include <stddef.h>
+#include <nvim/eval.h>
+
+#ifdef INCLUDE_GENERATED_DECLARATIONS
+# include "funcs.generated.h"
+#endif
+/*
+ * Function given to ExpandGeneric() to obtain the list of internal
+ * or user defined function names.
+ */
+char_u *get_function_name(expand_T *xp, int idx)
+{
+  static int intidx = -1;
+  char_u      *name;
+
+  if (idx == 0)
+    intidx = -1;
+  if (intidx < 0) {
+    name = get_user_func_name(xp, idx);
```

Comment:
Either reorder the function definitions or include `"nvim/eval/funcs.h"`.

Reply:
Better include funcs.h, header files are normally included in corresponding source files.